### PR TITLE
chore(monorepo): Clean up packages COMPASS-4704

### PR DESCRIPTION
**Pull request by gribnoysup**
In the scope of this ticket we made sure that some minimal amount of clean-up was applied to the packages so they are easier to work with. This list includes following changes

- Remove `mongodb-js-fmt`, corresponding .jsfmt config and package.json scripts using it. This library is deprecated, if we want to have automated code formatting, we should rather introduce it on the root level with other tool like `prettier` (create a ticket for that?)
- Remove `mongodb-js-precommit`, replace it with direct usage of `depcheck` and `eslint`. `mongodb-js-precommit` seems to be not well maintained for awhile now, breaks npm installs and doesn't bring enough value to fix it instead of removing it. In general precommit hooks now need to happen on the root level of the monorepo and not on the package, so separating precommit from lint/depcheck makes sense (`depcheck` currently only reports issues without throwing, we will do a clean-up separately in COMPASS-4772 post migration, when dependency versions are consolidated, this should make the process a little more straightforward, pre-commit hooks on the root level will be added back in COMPASS-4708)
- Remove all .github metadata and travis/azure ci configs. All github metadata / CI configuration should be at the root level now (new CI config will be added in the scope of COMPASS-4708)
- Fix all eslint errors that started to pop up after moving away from `mongodb-js-precommit`

TODO: 

- [ ] ~Make sure tests are passing in all the packages~ Doing in a separate PR

### CODE REVIEW

File: scripts/monorepo/with-progress.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const ora = require('ora');
+
+async function withProgress(text, fn, ...args) {
+  spinner = ora(text).start();
```
Comment (addaleax) [inline scripts/monorepo/with-progress.js:4]: TIL about `ora` :)

**Review by addaleax — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -10,25 +10,21 @@
   "shortcutFolderName": "MongoDB",
   "license": "SSPL",
   "scripts": {
-    "check": "lerna run check --scope mongodb-compass",
+    "check": "lerna run check --stream",
     "install": "lerna bootstrap --concurrency=1 --scope mongodb-compass",
     "start": "lerna run start --stream --scope mongodb-compass",
     "release": "cd packages/compass && npm run --silent release --",
-    "test": "lerna run test --stream --scope mongodb-compass"
+    "test": "lerna run test --stream"
```
Comment (gribnoysup) [inline package.json:17]: Let's bring the scope back for now, tests will to in a separate PR

```suggestion
    "test": "lerna run test --stream --scope mongodb-compass"
```

**Comment by gribnoysup** [comment]
As discussed on daily, I'm marking this ready for review. Most of this PR is autogenerated with a [script added here](https://github.com/mongodb-js/compass/pull/2173/files#diff-a5e10a8980bd6ae8fc1636aad3c7d5512fbfdbb94ccc35c32c16e479cbb01df6) another commit that might be of any interest I think is this one 55c2384eba69336824a77a53dee3c63d566f2a0c which contains all eslint fixes applied (99% of them is removing spaces, auto-closing react component tags, and changing var to let / const)

File: scripts/monorepo/with-progress.js (hunk 1 above)
Comment (gribnoysup) [reply scripts/monorepo/with-progress.js:4]: If I have to run something multiple times, I just had to have some pretty spinners to stare at at the very least 😆

**Pull request merged**